aten/src/ATen/native/native_functions.yaml

```diff
@@ -11119,6 +11119,14 @@
     CUDA: foreach_tensor_log2_cuda_
   autogen: _foreach_log2.out
 
+- func: _foreach_max(Tensor[] self) -> Tensor[]
+  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices
+  variants: function
+  dispatch:
+    CPU: foreach_tensor_max_slow
```

Comment:
Wait, sorry, what is the question?

Reply:
per offline discussion: change this key to CompositeExplicitAutograd.
Since other foreach ops can be upgraded as well, you can do that in a follow up.

Reply:
Okay!